src/compile/repeater.ts

```diff
@@ -57,32 +57,50 @@ function replaceRepeaterInFieldDef(fieldDef: ScaleFieldDef<Field>, repeater: Rep
   return fieldDef as ScaleFieldDef<string>;
 }
 
+function replaceRepeaterInChannelDef(channelDef: ChannelDef<Field>, repeater: RepeaterValue): ChannelDef<string> {
+  if (isConditionalDef(channelDef) && isFieldDef(channelDef.condition)) {
+    const fd = replaceRepeaterInFieldDef(channelDef.condition, repeater);
+    if (fd) {
+      return {
+        ...channelDef,
+        condition: fd
+      } as ChannelDef<string>;
+    } else {
+      return {
```

Comment:
better do
```
const {condition, ...channelDefWithoutCondition} = channelDef;
```

and 

`return channelDefWithoutCondition` 

as this will (1) not require `as` and (2) be more sustainable in long term should we add `datum` or additional property to a value def.

Reply:
Lol, I had exactly that and changed it back because I wanted to make it explicit that the second thing has to be a value.